### PR TITLE
Jcs/adhoc project

### PR DESCRIPTION
**Pull request by jSylvestre**
image

### CODE REVIEW

File: Harvest.Web/ClientApp/src/Projects/AdhocProject.tsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    // get rates so we can load up all expense types and info
+    const cb = async () => {
+      const response = await authenticatedFetch("/api/Rate/Active");
+
+      if (response.ok) {
+        const rates: Rate[] = await response.json();
+
+        getIsMounted() && setRates(rates);
+
+        // create default activity
```
Comment (srkirkland) [inline Harvest.Web/ClientApp/src/Projects/AdhocProject.tsx:77]: ?

File: Harvest.Web/Controllers/Api/ProjectController.cs [hunk 2]
```diff
@@ -185,5 +186,115 @@ public async Task<IActionResult> RefreshTotal(int projectId)
...
+                // Accounts will be auto-approved by quote approver
+                account.ApprovedById = currentUser.Id;
+                account.ProjectId= newProject.Id;
+                account.ApprovedOn = DateTime.UtcNow;
+                percentage += account.Percentage;
+                if (account.Percentage < 0)
+                {
+                    return BadRequest("Negative Percentage Detected");
+                }
+                newProject.Accounts.Add(account); //Don't need to specify projectId if I add it to the project?
```
Comment (srkirkland) [inline Harvest.Web/Controllers/Api/ProjectController.cs:264]: I believe yes, as long as you are adding to the array you don't need to set projectId, and thus you don't need to do this double-save thing, you can just do it all at once

Comment (jSylvestre) [reply Harvest.Web/Controllers/Api/ProjectController.cs:264]: Yeah... I had a different error at first and it wasn't working and I couldn't catch the exception. I'll change back.

File: Harvest.Web/ClientApp/src/Projects/AdhocProject.tsx (hunk 1 above)
Comment (jSylvestre) [reply Harvest.Web/ClientApp/src/Projects/AdhocProject.tsx:77]: @srkirkland  It is in the expense container:
https://github.com/ucdavis/Harvest/blob/main/Harvest.Web/ClientApp/src/Expenses/ExpenseEntryContainer.tsx#L95

**Review by srkirkland — APPROVED** [review]

**Pull request merged**